lib/server/auth.js

```diff
@@ -114,48 +136,87 @@ class DataAccess extends EventEmitter {
     });
     LOGGER.debug('ACCESS GROUP IDS: ' + agroupIds.join(' '));
 
-    let authurl = config.provide().get('authurl');
+    let auth_server = config.provide().get('auth_server');
+    let authurl = auth_server.url;
     let reqbody = {groups: agroupIds};
-    let reqopts = url.parse(authurl);
-    reqopts.method = 'POST';
+    let reqopts = {};
+    reqopts.url     = url.parse(authurl);
+    reqopts.method  = 'POST';
+    reqopts.headers = {
+      'Content-Type': 'application/json'
+    };
+
+    if ( authurl.startsWith(constants.SECURE_URI_PREFIX) ) {
```

Comment:
constants.SECURE_PROTOCOL (or replace other in config_helpers L13 with constants.SECURE_URI_PREFIX)